lib/core/src/Cardano/Wallet/Primitive/Types/UTxO/Gen.hs

```diff
@@ -0,0 +1,79 @@
+module Cardano.Wallet.Primitive.Types.UTxO.Gen
+    ( genUTxOSmall
+    , genUTxOLarge
+    , genUTxOLargeN
+    , shrinkUTxOSmall
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types.Tx
+    ( TxIn, TxOut )
+import Cardano.Wallet.Primitive.Types.Tx.Gen
+    ( genTxInLargeRange
+    , genTxInSmallRange
+    , genTxOutSmallRange
+    , shrinkTxInSmallRange
+    , shrinkTxOutSmallRange
+    )
+import Cardano.Wallet.Primitive.Types.UTxO
+    ( UTxO (..) )
+import Control.Monad
+    ( replicateM )
+import Test.QuickCheck
+    ( Gen, choose, frequency, shrinkList )
+import Test.QuickCheck.Extra
+    ( shrinkInterleaved )
+
+import qualified Data.Map as Map
+
+--------------------------------------------------------------------------------
+-- Small indices
+--------------------------------------------------------------------------------
+
+genUTxOSmall :: Gen UTxO
+genUTxOSmall = do
+    entryCount <- frequency
+        [ (1, pure 0)
+        , (1, pure 1)
+        , (32, choose (2, 64))
```

Comment:
Yes, this is what the QuickCheck `size` parameter is for.

Reply:
Yup (I don't want to change anything now though, as this is merely mirroring `UTxOIndex.Gen`)

Reply:
Sure. It's something we should fix though, so I tried fixing it... but found that there is a long chain of small/large variants which need to be changed. And so I gave up. :frowning_face: